### PR TITLE
minor improvements

### LINKED ISSUE #179: No use of `if` statement. if it will be written as of now.
I think we mistakenly added the extra statement at line 302
https://github.com/PolymathNetwork/polymath-core/blob/8027a0bc6ec44bd61d3869a1e4a9b6b6a2b2cfd1/contracts/Compliance.sol#L298-L302

### PR DESCRIPTION
**Pull request by satyamakgec**
fix #179 #168

### CODE REVIEW

File: flat/Template.sol [hunk 1]
```diff
@@ -115,6 +118,7 @@ contract Template is ITemplate {
         require(_details.length > 0 && _expires > now && _issuerJurisdiction.length > 0);
         require(_quorum > 0 && _quorum <= 100);
         require(_vestingPeriod > 0);
+        require(_fee > 0);
```
Comment (pabloruiz55) [inline flat/Template.sol]: Let's remove this so fee can be free.

File: flat/Template.sol [hunk 2]
```diff
@@ -138,12 +142,17 @@ contract Template is ITemplate {
         require(_allowedJurisdictions.length == _allowed.length);
         require(!finalized);
         for (uint i = 0; i < _allowedJurisdictions.length; ++i) {
+            if(!allowedJurisdictions[_allowedJurisdictions[i]] && _allowed[i])
```
Comment (pabloruiz55) [inline flat/Template.sol:144]: Hadn't I already added this and was merged?

**Review by pabloruiz55 — CHANGES_REQUESTED** [review]
Looks good, just left 2 items to review.

File: flat/Template.sol (hunk 2 above)
Comment (satyamakgec) [reply flat/Template.sol:144]: @pabloruiz55 this is flattened version of the contracts

File: flat/Template.sol (hunk 1 above)
Comment (satyamakgec) [reply flat/Template.sol]: So now we are also entertaining the zero fee ?

Comment (satyamakgec) [reply flat/Template.sol]: Done

File: flat/Template.sol (hunk 2 above)
Comment (pabloruiz55) [reply flat/Template.sol:144]: Right, nvm.

**Review by pabloruiz55 — APPROVED** [review]

**Pull request merged**